File: Tests/server/TextGenerator.cpp
#include "TextGenerator.h"

static constexpr size_t smallStringSize = 15;

TextGenerator::TextGenerator(std::string_view data) :
	data(data),
	offset(0)
{

}

std::string TextGenerator::generate()
{
	if (offset >= data.size())
	{
		return {};
	}

	std::string result = data.substr(offset, (std::min)(smallStringSize, data.size() - offset));

	offset += result.size();

	return result;
}

File: Tests/executors/CSharp/CoreExecutors/TextGenerator.cs
using Framework.Utility;
using System.Text;

namespace CoreExecutors
{
	public class TextGenerator(string data) : ChunkGenerator
	{
		private readonly string data = data;
		private int offset = 0;

		public override byte[] Generate(ref bool finished)
		{
			const int smallStringSize = 15;

			if (offset >= data.Length)
			{
				finished = true;

				return [];
			}

			string temp = data.Substring(offset, Math.Min(smallStringSize, data.Length - offset));

			offset += temp.Length;

			return Encoding.UTF8.GetBytes(temp);
		}
	}
}
